### PR TITLE
Bumblebee adjustments

### PR DESCRIPTION
**Pull request by elioschmutz**
@maethu 

Adjustments for the bumblebee-integration

### CODE REVIEW

File: docs/HISTORY.txt [hunk 1]
```diff
@@ -5,6 +5,14 @@ Changelog
 1.10.1 (unreleased)
 -------------------
 
+- Add adjustments to integrate bumblebee
+  - Trigger event dndUploadViewUpdated if drag&drop is finished
```
Comment (jone) [inline docs/HISTORY.txt]: this is not valid rst

File: ftw/file/utils.py [hunk 2]
```diff
@@ -25,3 +26,70 @@ def redirect_to_download_by_default(context):
...
+
+        if not site_props.getProperty('allowAnonymousViewAbout', False) \
+                and mt.isAnonymousUser():
+            return False
+        return True
+
+    @property
+    def author(self):
+        """ returns the fullname from creator or the id if there isn't a
+            fullname.
```
Comment (jone) [inline ftw/file/utils.py]: this comment is wrong, it returns a dict with more information :wink:

File: ftw/file/utils.py [hunk 3]
```diff
@@ -25,3 +26,70 @@ def redirect_to_download_by_default(context):
             request.other['disable_border'] = border_was_force_disabled
         if border_was_force_enabled:
             request.other['enable_border'] = border_was_force_enabled
+
+
+class FileMetadata(object):
+    """Handles some metadata of an object
```
Comment (jone) [inline ftw/file/utils.py]: can you better explain why this class exists?
this comment is not really helpful

File: ftw/file/utils.py [hunk 4]
```diff
@@ -25,3 +26,70 @@ def redirect_to_download_by_default(context):
...
+    """
+    def __init__(self, context):
+        self.context = context
+
+    @property
+    def show_author(self):
+        """Checks if the user is anonymous and is not allowAnonymousViewAbout.
```
Comment (jone) [inline ftw/file/utils.py]: I'd rather tell why I should use the property rather how it does things.
mayb ``True when the file author should be displayed``

**Comment by elioschmutz** [comment]
@jone 

:+1:

File: ftw/file/utils.py [hunk 5]
```diff
@@ -25,3 +26,90 @@ def redirect_to_download_by_default(context):
...
+    to get different pre-configured metadata of a file-object.
+    """
+    def __init__(self, context):
+        self.context = context
+
+    @property
+    def show_author(self):
+        """True when the file author should be displayed.
+
+        False: if visitor is anonymous an siteproperty
```
Comment (jone) [inline ftw/file/utils.py]: `anD`

**Comment by jone** [comment]
:+1:

**Comment by Tschanzt** [comment]
@jone rebased.

**Pull request merged**